### PR TITLE
Cgroup execjob_end_handler should also call cgroup.delete()

### PR DESCRIPTION
**Pull request by minghui-liu**
#### Bug/feature Description
Cgroup execjob_epilogue event could fail or skip and leave cgroup files for a job behind at the time when execjob_end handler is running.

#### Affected Platform(s)
All platforms that support cgroup hook

#### Cause / Analysis / Design
Currently cgroup hook relies on `execjob_epilogue_handler` to clean up cgroup files of a job. There are cases that `execjob_epilogue_handler` is not run. For example, an execjob_epilogue hook before the cgroup hook rejected the job. By the time `execjob_end_handler` runs the files will still be there.

#### Solution Description
Add a call to `cgroup.delete()` in `execjob_end_handler` in case cgroup execjob_epilogue event failed to clean up.

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by minghui-liu** [comment]
Here is the [PTL test log](https://github.com/PBSPro/pbspro/files/1753514/cgroup_test_output.txt)

File: src/hooks/cgroups/pbs_cgroups.PY [hunk 1]
```diff
@@ -752,6 +752,8 @@ class HookUtils(object):
         """
         pbs.logmsg(pbs.EVENT_DEBUG4, "%s: Method called" % caller_name())
         # The cgroup was already deleted in the epilogue handler.
+        # Delete it again regardless, just in case
```
Comment (bhroam) [inline src/hooks/cgroups/pbs_cgroups.PY]: How about: "The cgroup is usually deleted in the execjob_epilogue hook.  There are certain corner cases where that hook event will not be called.  We delete it again here to make sure we catch those.  It'll be a noop if the cgroup is already deleted"

File: test/tests/functional/pbs_cgroups_hook.py [hunk 2]
```diff
@@ -36,6 +36,7 @@
 # trademark licensing policies.
 
 from tests.functional import *
+import os
```
Comment (mike0042) [inline test/tests/functional/pbs_cgroups_hook.py]: The os module is already loaded. No need to do so here.

Comment (minghui-liu) [reply test/tests/functional/pbs_cgroups_hook.py]: Removed. Thanks.

**Comment by minghui-liu** [comment]
Thanks for the reviews. I have squashed and rebased.
